File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 8/23/2025, 3:51:52 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int sum;
        vector<int> vect;
        for(int i=0;i<nums.size();i++)
        {
            if(i==nums.size()-1)
                break;
            for(int j=i+1;j<nums.size();j++)
            {
                sum=nums[i]+nums[j];
                if(sum==target)
                {
                    vect.push_back(i);
                    vect.push_back(j);
                    break;
                }
            }
        }   
        return vect;
    }
};

File: 1-TwoSum/1-TwoSum.py
# Last updated: 8/23/2025, 3:51:51 PM
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        # for i in range(len(nums)):
        #     for j in range(i+1, len(nums)):
        #         if nums[i] + nums[j] == target:
        #             return [i, j]
        # return []

    
        seen = {}
        for i, val in enumerate(nums):
            diff = target - nums[i]
            if diff in seen:
                return [i, seen[diff]]
            seen[val] = i
        return []
            
        